tolerance in Series and DataFrame.reindex does not like Timedelta

# Describe the bug
In practice and [in the documentation](https://github.com/pandas-dev/pandas/issues/62010#issuecomment-3140783624), `tolerance` in `reindex` of `Series` and `DataFrame` supports `Timedelta`, which is not recognised in the stubs.

# To Reproduce
## Minimal reproducible script
```py
import pandas as pd

sr = pd.Series([1, 2], pd.to_datetime(["2023-01-01", "2023-01-02"]))
sr.reindex(index=pd.to_datetime(["2023-01-02", "2023-01-03"]), method="ffill", tolerance=pd.Timedelta("1D"))

df = sr.to_frame()
df.reindex(index=pd.to_datetime(["2023-01-02", "2023-01-03"]), method="ffill", tolerance=pd.Timedelta("1D"))
```
## Indicate which type checker you are using
Both
## Show the error message received from that type checker while checking your example.
### `mypy`
```
attempt.py:4: error: Argument "tolerance" to "reindex" of "Series" has incompatible type "Timedelta"; expected "float | None"  [arg-type]
attempt.py:7: error: Argument "tolerance" to "reindex" of "DataFrame" has incompatible type "Timedelta"; expected "float | None"  [arg-type]
```
### `pyright`
```
  attempt.py:4:90 - error: Argument of type "Timedelta" cannot be assigned to parameter "tolerance" of type "float | None" in function "reindex"
    Type "Timedelta" is not assignable to type "float | None"
      "Timedelta" is not assignable to "float"
      "Timedelta" is not assignable to "None" (reportArgumentType)
  attempt.py:7:90 - error: Argument of type "Timedelta" cannot be assigned to parameter "tolerance" of type "float | None" in function "reindex"
    Type "Timedelta" is not assignable to type "float | None"
      "Timedelta" is not assignable to "float"
      "Timedelta" is not assignable to "None" (reportArgumentType)
2 errors, 0 warnings, 0 informations
```

# Please complete the following information:
## OS
Windows
## OS Version
11
## python version
3.12.10
## version of type checker
- mypy: 1.17.0
- pyright: 1.1.403
## version of installed `pandas-stubs`
Current `main` branch

# Additional context
None

--- a/pandas-stubs/core/frame.pyi
+++ b/pandas-stubs/core/frame.pyi
@@ -816,7 +816,7 @@ class DataFrame(NDFrame, OpsMixin, _GetItemHack):
         level: int | _str = ...,
         fill_value: Scalar | None = ...,
         limit: int | None = None,
-        tolerance: float | None = ...,
+        tolerance: float | Timedelta | None = ...,
     ) -> Self: ...
     @overload
     def rename(

--- a/pandas-stubs/core/series.pyi
+++ b/pandas-stubs/core/series.pyi
@@ -1234,7 +1234,7 @@ class Series(IndexOpsMixin[S1], NDFrame):
         level: int | _str | None = None,
         fill_value: Scalar | None = None,
         limit: int | None = None,
-        tolerance: float | None = None,
+        tolerance: float | Timedelta | None = None,
     ) -> Series[S1]: ...
     def filter(
         self,

--- a/tests/series/test_series.py
+++ b/tests/series/test_series.py
@@ -3826,8 +3826,30 @@ def test_series_int_float() -> None:
 
 
 def test_series_reindex() -> None:
+    """Test Series.reindex without any arguments and with tolerance."""
     s = pd.Series([1, 2, 3], index=[0, 1, 2])
     check(assert_type(s.reindex([2, 1, 0]), "pd.Series[int]"), pd.Series, np.integer)
+    check(
+        assert_type(
+            s.reindex([2, 1, 0], method="backfill", tolerance=1), "pd.Series[int]"
+        ),
+        pd.Series,
+        np.integer,
+    )
+
+    sr = pd.Series([1, 2], pd.to_datetime(["2023-01-01", "2023-01-02"]))
+    check(
+        assert_type(
+            sr.reindex(
+                index=pd.to_datetime(["2023-01-02", "2023-01-03"]),
+                method="ffill",
+                tolerance=pd.Timedelta("1D"),
+            ),
+            "pd.Series[int]",
+        ),
+        pd.Series,
+        np.integer,
+    )
 
 
 def test_series_reindex_like() -> None:

--- a/tests/test_frame.py
+++ b/tests/test_frame.py
@@ -3178,9 +3178,29 @@ def test_frame_stack() -> None:
 
 
 def test_frame_reindex() -> None:
+    """Test DataFrame.reindex with different arguments."""
     # GH 84
     df = pd.DataFrame({"a": [1, 2, 3]}, index=[0, 1, 2])
     check(assert_type(df.reindex([2, 1, 0]), pd.DataFrame), pd.DataFrame)
+    check(
+        assert_type(df.reindex([2, 1, 0], method="pad", tolerance=2.3), pd.DataFrame),
+        pd.DataFrame,
+    )
+
+    # GH1307
+    sr = pd.Series([1, 2], pd.to_datetime(["2023-01-01", "2023-01-02"]))
+    df = sr.to_frame()
+    check(
+        assert_type(
+            df.reindex(
+                index=pd.to_datetime(["2023-01-02", "2023-01-03"]),
+                method="ffill",
+                tolerance=pd.Timedelta("1D"),
+            ),
+            pd.DataFrame,
+        ),
+        pd.DataFrame,
+    )
 
 
 def test_frame_reindex_like() -> None:

